app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -0,0 +1,24 @@
+package com.intive.picover.profile.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.auth.repository.AuthRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ProfileViewModel @Inject constructor(
+	private val authRepository: AuthRepository,
+) : ViewModel() {
+
+	fun onLogoutClick() {
+		authRepository.logout()
```

Comment:
It seems that the logout does not return us to the sign-in page, only deleting the account does:

https://github.com/intive/Picover/assets/33199010/f25091dc-7105-4447-b422-c50150430d9c

Reply:
That's not an issue of the implementation but the fact that you saved the credentials using "Smart Lock". Without saving the password it works as expected, however when the password is saved to "Smart Lock" then after logout you are automatically logged in. We can think of it later, maybe there is some way to tweak it, or we can disable "Smart Lock" (I've seen option for that).

Reply:
Ok. Good to know. Thanks.